### PR TITLE
Improve dt_filesocket transport.

### LINKED ISSUE #235: Improve dt_filesocket implementation
The dt_filesocket debug transport in the sapmachine11 repository needs a few improvemts:

- A jtreg test
- A mechanism to avoid leaving file socket files behind
- Checking the connected user and not just relying on the file permissions. This is needed since on some platforms the connection permission is taken form the permission of the directory in which the domain sockets is created. Additionally it will be easier later to support abstract domain socket on unix later, since they cannot rely on file permissions at all.
@

### PR DESCRIPTION
**Pull request by schmelter-sap**
This pull requests includes various improvements for the dt_filesocket transport library.

fixes #235

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

File: src/jdk.jdwp.agent/share/native/libdt_filesocket/fileSocketTransport.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2018, 2019, Oracle and/or its affiliates. All rights reserved.
```
Comment (RealCLanger) [inline src/jdk.jdwp.agent/share/native/libdt_filesocket/fileSocketTransport.c]: Unnecessary update of copyright year.

File: src/jdk.jdwp.agent/share/native/libdt_filesocket/fileSocketTransport.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2018, 2019, Oracle and/or its affiliates. All rights reserved.
```
Comment (RealCLanger) [inline src/jdk.jdwp.agent/share/native/libdt_filesocket/fileSocketTransport.h]: Unnecessary update of copyright year.

File: test/jdk/com/sun/jdi/FileSocketTransportTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8214892
```
Comment (RealCLanger) [inline test/jdk/com/sun/jdi/FileSocketTransportTest.java]: For that bug there is another test. This line should probably be deleted.

File: test/jdk/com/sun/jdi/FileSocketTransportTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+ * @requires os.arch != "aarch64"
+ * @summary Test that the onjcmd option of the jdwp agent works.
```
Comment (RealCLanger) [inline test/jdk/com/sun/jdi/FileSocketTransportTest.java]: I think you rather test basic functionality of debugging via filesockets, right?

File: src/jdk.jdwp.agent/share/native/libdt_filesocket/fileSocketTransport.h (hunk 2 above)
Comment (schmelter-sap) [reply src/jdk.jdwp.agent/share/native/libdt_filesocket/fileSocketTransport.h]: OK

File: test/jdk/com/sun/jdi/FileSocketTransportTest.java (hunk 3 above)
Comment (schmelter-sap) [reply test/jdk/com/sun/jdi/FileSocketTransportTest.java]: Removed

File: test/jdk/com/sun/jdi/FileSocketTransportTest.java (hunk 4 above)
Comment (schmelter-sap) [reply test/jdk/com/sun/jdi/FileSocketTransportTest.java]: Fixed

File: src/jdk.jdwp.agent/unix/native/libdt_filesocket/fileSocketTransport_md.c [hunk 5]
```diff
@@ -29,127 +29,312 @@
...
+}
+
+static void cleanupStaleDefaultSockets() {
+    static char prefix[UNIX_PATH_MAX];
+    char const* tmpdir = getTempdir();
+    DIR* dir = opendir(tmpdir);
+
+    if (dir != NULL) {
+        struct dirent* ent;
+        int prefix_len = snprintf(prefix, sizeof(prefix), "sapmachine_dt_filesocket_%lld_", (long long) geteuid());
```
[the remark this answers is not in the document]
Comment (schmelter-sap) [reply src/jdk.jdwp.agent/unix/native/libdt_filesocket/fileSocketTransport_md.c]: Done

File: src/jdk.jdwp.agent/unix/native/libdt_filesocket/fileSocketTransport_md.c [hunk 6]
```diff
@@ -29,127 +29,312 @@
...
+                    long long pid = strtoll(ent->d_name + prefix_len, &pid_end, 10);
+
+                    if ((*pid_end == '_') && (pid != 0)) {
+                        char* guid_end;
+                        strtoll(pid_end + 1, &guid_end, 10);
+
+                        if ((pid_end[1] != '\0') && (*guid_end == '\0')) {
+                            errno = 0;
+
+                            if ((pid == (long long) getpid()) || ((kill((pid_t) pid, 0) == -1) && (errno = ESRCH))) {
```
[the remark this answers is not in the document]
Comment (schmelter-sap) [reply src/jdk.jdwp.agent/unix/native/libdt_filesocket/fileSocketTransport_md.c]: Fixed

File: src/jdk.jdwp.agent/unix/native/libdt_filesocket/fileSocketTransport_md.c [hunk 7]
```diff
@@ -29,127 +29,312 @@
...
+            return;
+        }
+
+        other_user = cred_info.euid;
+        other_group = cred_info.egid;
+#elif defined(__sun)
+        ucred_t* cred_info = NULL;
+
+        if (getpeerucred(handle, &cred_info) == -1) {
+            logAndCleanupFailedAccept("Failed to peer credientials of file socket", name);
```
[the remark this answers is not in the document]
Comment (schmelter-sap) [reply src/jdk.jdwp.agent/unix/native/libdt_filesocket/fileSocketTransport_md.c]: Fixed

**Comment by SapMachine** [comment]
Hello @schmelter-sap, this pull request fulfills all formal requirements.

**Pull request merged**